### PR TITLE
[wip] client

### PR DESCRIPTION
**Pull request by dzamoshchin**
currently testing with
```
cargo test main -- --nocapture
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -30,8 +30,6 @@ tracing = "0.1.40"
 tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
 axum = "0.7.5"
 thiserror = "1.0.59"
-
-[dev-dependencies]
```
Comment (ratoru) [inline Cargo.toml:34]: Why? I don't think we were using them in the production code, no? I thought they were only for `cargo test`

File: src/lib.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ use crate::store::Response;
 
 pub mod app;
 pub mod carp;
-pub mod client;
+pub mod node;
```
Comment (ratoru) [inline src/lib.rs]: Idk if this is gonna conflict with the `struct Node`?

File: tests/client.rs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    let initial_load = 1.0 / num_clusters as f32;
+    let mut carp_ring = Carp::new(
+        all_nodes.iter().map(|cluster| {
+            let (leader_addr, _) = &cluster[0];
+            (leader_addr.clone(), initial_load)
+        }).collect(),
+        0,
+    );
+
+    // Display the CARP ring nodes
```
Comment (ratoru) [inline tests/client.rs]: We could implement the Display trait for Carp in `carp.rs` if you want to print out `Carp`

**Review by ratoru — COMMENTED** [review]
Nice! Thanks for seperating client and raftnode client.

File: tests/client.rs (hunk 3 above)
Comment (dzamoshchin) [reply tests/client.rs]: will do!

File: Cargo.toml (hunk 1 above)
Comment (dzamoshchin) [reply Cargo.toml:34]: right, i will revert this

**Comment by dzamoshchin** [comment]
inside of ``client`` can now test with admin launching the ring of clusters:
```
cargo run --bin admin
```

and in a separate terminal, test by getting hash ring:
```
curl "127.0.0.1:31301/api/get_hash_ring" -H "Content-Type: application/json" -d ''
```

File: Cargo.toml (hunk 1 above)
Comment (ratoru) [reply Cargo.toml:34]: Did you forget this?

File: src/bin/admin.rs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+use std::fs;
+use std::error::Error;
+
+use toml;
+use serde::Deserialize;
+use serde_json;
+
+#[derive(Deserialize)]
+struct Config {
```
Comment (ratoru) [inline src/bin/admin.rs:19]: Should we add IP addresses to Config struct instead of hard-coding?

File: src/bin/admin.rs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    let serialized_all_nodes = serde_json::to_string(&all_nodes)?;
+    fs::write("all_nodes.json", serialized_all_nodes)?;
+
+    println!("Application is running. Press Ctrl+C to exit.");
+    tokio::signal::ctrl_c().await.expect("Failed to listen for ctrl_c");
+   
+    for handle in handles {
+        handle.abort();
```
Comment (ratoru) [inline src/bin/admin.rs]: Does this shut down the clusters?

**Review by ratoru — CHANGES_REQUESTED** [review]
Did `cargo test` pass?

File: src/bin/admin.rs (hunk 4 above)
Comment (dzamoshchin) [reply src/bin/admin.rs:19]: this is a good idea, will do

**Pull request merged**